src/checks/obsolete-record/match-filename-slug.ts

```diff
@@ -0,0 +1,21 @@
+import { basename } from 'path';
+import { ObsoleteRecordCheck } from '../../types/checks';
+
+const check: ObsoleteRecordCheck = {
+    id: 'match-filename-slug',
+    desc: 'Obsolete record filenames have to match their slug.',
+    url: 'https://github.com/datenanfragen/data#handling-obsolete-records',
+    severity: 'ERROR',
+    run: (json, ctx) => {
+        const filename = basename(ctx.file_path);
+        if (json['slug'] + '.json' !== filename)
+            return {
+                message: `Filename \`${filename}\` does not match slug \`${json.slug}\`. They need to match and be the same as the original record.`,
+                json_pointer: '/slug',
+                suggestions: [basename(filename, '.json')],
```

Comment:
```suggestion
        const filename = basename(ctx.file_path, '.json');
        if (json['slug']!== filename)
            return {
                message: `Filename \`${filename}\` does not match slug \`${json.slug}\`. They need to match and be the same as the original record.`,
                json_pointer: '/slug',
                suggestions: [filename],
```
Why not call `basename()` only once? This code looks more readable to me.

Reply:
I think _as-is_, your suggestion is less readable because `filename` doesn't actually hold the filename anymore. If we rename it to `filenameWithoutExtension` or something like that, I fine with either option. I just mostly copied this from https://github.com/datenanfragen/data/blob/fb5cbf7e2566d12a1288ad2fa03caa0dfa6c86d9/src/checks/record/match-filename-slug.ts, so if we do decide to change it, we should also propagate those changes there.